tests/test_feedexport.py

```diff
@@ -186,6 +187,158 @@ def test_store(self):
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, None)
+
+        def _defer(f, *args, **kwargs):
+            return f(*args, **kwargs)
+
+        with mock.patch('botocore.client.BaseClient._make_api_call') as m:
+            with mock.patch('twisted.internet.threads.deferToThread',
+                            new=_defer):
+                storage.store(BytesIO(b'test file'))
+
+            operation_name, api_params = m.call_args[0]
+            self.assertEqual(operation_name, 'PutObject')
+            self.assertNotIn('ACL', api_params)
+
+    def test_store_botocore_with_acl(self):
+        storage = S3FeedStorage(
+            's3://mybucket/export.csv',
+            'access_key',
+            'secret_key',
+            'custom-acl'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, 'custom-acl')
+
+        def _defer(f, *args, **kwargs):
+            return f(*args, **kwargs)
+
+        with mock.patch('botocore.client.BaseClient._make_api_call') as m:
+            with mock.patch('twisted.internet.threads.deferToThread',
+                            new=_defer):
+                storage.store(BytesIO(b'test file'))
+
+            operation_name, api_params = m.call_args[0]
+            self.assertEqual(operation_name, 'PutObject')
+            self.assertEqual(api_params.get('ACL'), 'custom-acl')
+
+    def test_store_not_botocore_without_acl(self):
+        storage = S3FeedStorage(
+            's3://mybucket/export.csv',
+            'access_key',
+            'secret_key',
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, None)
+
+        storage.is_botocore = False
+        storage.connect_s3 = mock.MagicMock()
+        self.assertFalse(storage.is_botocore)
+
+        def _defer(f, *args, **kwargs):
+            return f(*args, **kwargs)
+
+        with mock.patch('twisted.internet.threads.deferToThread', new=_defer):
+            storage.store(BytesIO(b'test file'))
+
+        conn = storage.connect_s3(*storage.connect_s3.call_args)
+        bucket = conn.get_bucket(*conn.get_bucket.call_args)
+        key = bucket.new_key(*bucket.new_key.call_args)
+        self.assertNotIn(
+            dict(policy='custom-acl'),
+            key.set_contents_from_file.call_args
+        )
+
+    def test_store_not_botocore_with_acl(self):
+        storage = S3FeedStorage(
+            's3://mybucket/export.csv',
+            'access_key',
+            'secret_key',
+            'custom-acl'
+        )
+        self.assertEqual(storage.access_key, 'access_key')
+        self.assertEqual(storage.secret_key, 'secret_key')
+        self.assertEqual(storage.acl, 'custom-acl')
+
+        storage.is_botocore = False
+        storage.connect_s3 = mock.MagicMock()
```

Comment:
Maybe `mock.create_autospec()` is better than `MagicMock` because it could help catching errors if the API changes or if there is a typo in the test/implementation.

Reply:
I don't know. I decided to go with this simple method because I just would like to guarantee we're calling the `put_object` and `new_key` methods with `ACL` and `policy` arguments respectively.

If we change the method names, the code would break because the mocks wouldn't have been called from the test.

Let me know if I am missing anything.

Reply:
It is fine for me, the main concern was _if the API (method names) change, the unit test won't catch these errors and we'll notice them in production or so. But, if we `create_autospec` it will create a mock with the same methods available in the API. So, if the API changes and we call a method that does not exist anymore, we'll get an error._
Again, that is just a thought and I was collecting some ideas as well :)